src/components/Clock.tsx

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-console */
+import React from 'react';
+
+type Props = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+type State = {
+  date: Date;
+};
+
+export default class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  componentDidMount() {
```

Comment:
if subscribe once, you have to unsubscribe from interval in `componentWillUnmount` as well